async-nats/src/jetstream/context.rs

```diff
@@ -49,4 +54,25 @@ impl Context {
 
         Ok(response)
     }
+
+    pub async fn add_stream<S>(&mut self, stream_config: S) -> Result<StreamInfo, Error>
+    where
+        StreamConfig: From<S>,
+    {
+        let config: StreamConfig = stream_config.into();
+        if config.name.is_empty() {
+            return Err(Box::new(io::Error::new(
+                ErrorKind::InvalidInput,
+                "the stream name must not be empty",
+            )));
+        }
+        let subject: String = format!("{}.STREAM.CREATE.{}", self.prefix, config.name);
```

Comment:
Compiler can figure this one out can't it?

```suggestion
        let subject = format!("{}.STREAM.CREATE.{}", self.prefix, config.name);
```